libhfnetdriver/netdriver.c

```diff
@@ -134,11 +140,12 @@ static void netDriver_bindToRndLoopback(int sock, sa_family_t sa_family) {
     }
 }
 
-static int netDriver_sockConnAddr(const struct sockaddr *addr, socklen_t socklen) {
-    int sock = socket(addr->sa_family, SOCK_STREAM, 0);
+static int netDriver_sockConnAddr(
+    const struct sockaddr *addr, socklen_t socklen, int type, int protocol) {
+    int sock = socket(addr->sa_family, type, protocol);
     if (sock == -1) {
-        PLOG_W("socket(type=%d for dst_addr='%s', SOCK_STREAM, 0)", addr->sa_family,
-            files_sockAddrToStr(addr, socklen));
+        PLOG_W("socket(family=%d for dst_addr='%s', type=%d, protocol=%d)", addr->sa_family,
+            files_sockAddrToStr(addr, socklen), type, protocol);
         return -1;
     }
     if (addr->sa_family == AF_INET || addr->sa_family == AF_INET6) {
```

Comment:
With the introduction of arbitrary socket types perhaps the options should be set on all sockets and the warning/error only be emitted if it's supported (`ENOPROTOOPT`?) .

Reply:
Good catch. Done.